assets/js/api/elasticSearch.js

```diff
@@ -0,0 +1,148 @@
+import elasticsearch from 'elasticsearch';
+import { buildFilter, buildSort } from './elasticSearchUtils.js';
+import { SOFTWARE_OPEN, SOFTWARE_REUSE } from '../utils/constants.js';
+import { lang } from '../utils/l10n.js';
+
+const client = new elasticsearch.Client({
+  // eslint-disable-next-line no-undef
+  host: process.env.ELASTICSEARCH_FRONTEND_URL,
+});
+
+export const querySoftware = async ({ type, searchValue, filters, sortBy, from, size }) => {
+  const must = [{ term: { _type: 'software' } }];
+  if (searchValue) {
+    must.push({
+      multi_match: {
+        query: searchValue,
+        fields: [
+          'publiccode.name^3',
+          `publiccode.description.${lang}.localizedName^3`,
+          `publiccode.description.${lang}.shortDescription^2`,
+          `publiccode.description.${lang}.longDescription`,
+        ],
+      },
+    });
+  }
+
+  if (type === SOFTWARE_REUSE) {
+    must.push({ exists: { field: 'publiccode.it.riuso.codiceIPA' } });
+  }
+
+  const must_not = [];
+
+  if (type === SOFTWARE_OPEN) {
+    must_not.push(
+      { exists: { field: 'publiccode.it.riuso.codiceIPA' } },
+      { match: { 'publiccode.indendedAudience.unsupportedCountries': 'it' } }
+    );
+  }
+
+  const query = {
+    bool: {
+      filter: buildFilter(filters),
+      must,
+      must_not,
+    },
+  };
+  const results = await executeQuery({ query, sort: buildSort(sortBy), from, size });
+  return results;
```

Comment:
Can't we just `return await executeQuery({ query, sort: buildSort(sortBy), from, size });` ?

Reply:
Yes, we should. I have already changed it in the branch feat/eslint https://github.com/italia/developers.italia.it/blob/feat/eslint/assets/js/api/elasticSearch.js#L47

I would open a PR after merging this one in master